vignettes/hys_law.Rmd

```diff
@@ -0,0 +1,197 @@
+
+- Elevated alanine aminotransaminase (ALT) or aspartate aminotransaminase (AST) by 3-times or greater of the upper limit of normal.
+- Elevated serum total bilirubin (BILI) by 2-times or greater within a window of time, ~14 days after the elevated ALT/AST event.
+- No other reason to explain these increased lab values like preexisting liver disease.
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Creating the Appropriate `CRITX` and `CRITXFL` Columns](#columns)
+* [Subsetting by `LBTESTCD` and Joining by Potential Events](#joins)
+* [How to Create New Parameters and Rows](#newparams)
+* [Conclusion](#conclusion)
+
+## Read in Data {#readdata}
+
+We can use a mixture of `{admiral}` and basic `{tidyverse}` functions to assist us in creating an ADLBHY dataset. In the walkthrough below we will use the ADLB dataset created from the call `use_ad_template("adlb")`. Due to the size of the dataset, we only included the following USUBJID's: 
+
+* 01-701-1015, 01-701-1023, 01-701-1028, 01-701-1033, 01-701-1034, 01-701-1047, 01-701-1097, 01-705-1186, 01-705-1292, 01-705-1310, 01-708-1286. 
+
+```{r, echo = FALSE}
+adlb <- admiral_adlb
+head(adlb) %>%
+  dataset_vignette()
+```
+
+
+## Creating the Appropriate CRITX and CRITXFL Columns {#columns}
+
+A standard convention of ADLBHY datasets, are various `CRITX` and `CRITXFL` columns to describe the conditions necessary to reach that particular criterion of Hy's Law and the actual flag itself to indicate whether or not the condition was reached.
+
+Using `mutate()`, `call_derivation()` and `derive_var_merged_exist_flag()`, we can create these columns that indicate the the 3-fold or greater than upper limit of normal of ALT/AST and the 2-fold or greater than upper limit of normal of BILI.
+
+To increase visibility and for simplicity, we will retain only columns that are relevant to a Hy's Law analysis for now. 
+
+```{r}
+adlb_annotated <- adlb %>%
+  filter(PARAMCD %in% c("AST", "ALT", "BILI") & is.na(DTYPE)) %>%
+  mutate(CRIT1 = case_when(
+    PARAMCD == "AST" ~ "AST >=3xULN",
+    PARAMCD == "ALT" ~ "ALT >=3xULN",
+    PARAMCD == "BILI" ~ "BILI >=2xULN"
+  )) %>%
+  call_derivation(
+    .,
+    derivation = derive_var_merged_exist_flag,
+    dataset_add = .,
+    by_vars = exprs(USUBJID, LBSEQ, PARAMCD, ADT),
+    variable_params = list(
+      params(
+        new_var = CRIT1FL,
+        condition = (
+          (AVAL / ANRHI >= 3 & PARAMCD %in% c("AST", "ALT")) |
+            (AVAL / ANRHI >= 2 & PARAMCD == "BILI")
+        )
+      )
+    )
+  ) %>%
```

Comment:
It depends on the definition of the flags. In the example in the comment, the flags for AST elevations are merged to all parameters. You can't do this with `mutate()`.
For `CRITyFL` it is different because for the derivation you can use only values from the current records. Therefore you can always implement it via `mutate()`.